doc/backends.txt

```diff
@@ -178,6 +178,151 @@ could as well pass ``logic.o``) we could be passing switches to link any other
 static C library.
 
 
+C++ invocation example
+~~~~~~~~~~~~~~~~~~~~
+
+This example shows calling a C++ compiled dll and nim source, using the
+windows platform.
+
+Create a ``twoStor.hpp`` header file with the following content (and note that
+the EXPORT macro ensures that the generated DLL file exports the StorTwo class):
+
+.. code-block:: cpp
+
+  #ifdef MAKEDLL
+  #  define EXPORT __declspec(dllexport)
+  #else
+  #  define EXPORT __declspec(dllimport)
+  #endif
+
+  class EXPORT StorTwo
+  {
+  public:
+    EXPORT StorTwo(int a, int b);
+
+    void EXPORT setVals(int a, int b);
+    void EXPORT showVals();
+
+    int x, y;
+  };
+
+  StorTwo* EXPORT newStorTwo(int a, int b);
+
+Create a ``twoStor.cpp`` file with the following content:
+
+.. code-block:: cpp
+
+  #include "storTwo.hpp"
+  #include <stdio.h>
+
+  StorTwo* EXPORT newStorTwo(int a, int b)
+  {
+    StorTwo* t = new StorTwo(a, b);
+    return t;
+  }
+
+  EXPORT StorTwo::StorTwo(int a, int b)
+  {
+    this->x = a;
+    this->y = b;
+  }
+
+  void EXPORT StorTwo::setVals(int a, int b)
+  {
+    this->x = a;
+    this->y = b;
+  }
+
+  void EXPORT StorTwo::showVals(void)
+  {
+    printf("Values: %i, %i\n", this->x, this->y);
+  }
+
+  int main(void)
+  {
+    StorTwo t(1,2);
+    t.showVals();
+    t.setVals(23, 42);
+    t.showVals();
+  }
+
+Create a ``testTwoStor.nim`` file with the following content:
+
+
+.. code-block:: cpp
+
+  const
+    # location relative to nimcache directory
+    # if header is not already in the compiler include path
+    hdr = "../storTwo.hpp"
+
+  {.passL: "storTwo.dll".}
```

Comment:
That's not how we do interfacing with DLLs.

Reply:
Excellent.  It obviously needs clarifying, so that misunderstandings like this don't happen  :-)

Do I close this PR, and raise an issue for someone add an example?

Reply:
I will eventually merge it and clean it up. Marking as "postponed".